cardano-node/src/Cardano/Node/Configuration/Topology.hs

```diff
@@ -35,98 +36,83 @@ import           Ouroboros.Consensus.Util.Condense (Condense (..))
-    IP.IPv4 ipv4 -> SockAddrInet  naPort   (IP.toHostAddress  ipv4)
-    IP.IPv6 ipv6 -> SockAddrInet6 naPort 0 (IP.toHostAddress6 ipv6) 0
+nodeAddressToSockAddr :: NodeAddress -> SockAddr
+nodeAddressToSockAddr (NodeAddress addr port) =
+  case addr of
+    IP.IPv4 ipv4 -> SockAddrInet port $ IP.toHostAddress ipv4
+    IP.IPv6 ipv6 -> SockAddrInet6 port 0 (IP.toHostAddress6 ipv6) 0
 
 nodeAddressInfo :: NodeAddress -> AddrInfo
-nodeAddressInfo na@NodeAddress {naHostAddress}
-    = AddrInfo {
-        addrFlags      = addrFlags defaultHints
-      , addrFamily     = case naHostAddress of
-                          IP.IPv4{} -> AF_INET
-                          IP.IPv6{} -> AF_INET6
-      , addrSocketType = Stream
-      , addrProtocol   = addrProtocol defaultHints
-      , addrAddress    = nodeAddressToSockAddr na
-      , addrCanonName  = Nothing
-    }
+nodeAddressInfo na@(NodeAddress hostAddr _) =
+  AddrInfo { addrFlags = addrFlags defaultHints
+           , addrFamily = case hostAddr of
+                            IP.IPv4 _ -> AF_INET
+                            IP.IPv6 _ -> AF_INET6
+           , addrSocketType = Stream
+           , addrProtocol = addrProtocol defaultHints
+           , addrAddress = nodeAddressToSockAddr na
+           , addrCanonName = Nothing
+           }
 
 -- | Domain name with port number
 --
-data RemoteAddress = RemoteAddress {
-    raAddress :: !String
+data RemoteAddress = RemoteAddress
+  { raAddress :: !String
   -- ^ either a dns address or ip address
-  , raPort    :: !PortNumber
+  , raPort :: !PortNumber
   -- ^ port number of the destination
   , raValency :: !Int
-  -- ^ if a dns address is given valency governs to how many resolved ip addresses
+  -- ^ if a dns address is given valency governs
+  -- to how many resolved ip addresses
   -- should we maintain acctive (hot) connection;
-  -- if an ip address is given valency is used as a boolean value, @0@ means to
-  -- ignore the address;
-  }
-  deriving (Eq, Ord, Show)
+  -- if an ip address is given valency is used as
+  -- a boolean value, @0@ means to ignore the address;
+  } deriving (Eq, Ord, Show)
 
 
 -- | Parse 'raAddress' field as an IP address; if it parses and the valency is
 -- non zero return corresponding NodeAddress.
 --
-remoteAddressToNodeAddress
-  :: RemoteAddress
-  -> Maybe NodeAddress
-remoteAddressToNodeAddress RemoteAddress {raAddress, raPort, raValency} =
-    case readMaybe raAddress of
+remoteAddressToNodeAddress:: RemoteAddress-> Maybe NodeAddress
+remoteAddressToNodeAddress (RemoteAddress addrStr port val) =
+    case readMaybe addrStr of
       Nothing -> Nothing
-      Just naHostAddress | raValency /= 0 -> Just (NodeAddress {
-                                                      naHostAddress
-                                                    , naPort = raPort
-                                                    })
-                         | otherwise      -> Nothing
+      Just addr -> if val /= 0
```

Comment:
I think this somewhat contradicts https://github.com/input-output-hk/cardano-node/blob/master/STYLE.md#if-then-else-clauses

> Generally, guards and pattern matches should be preferred over if-then-else clauses, where possible.

Reply:
Hmm...I genuinely felt like the `if then else` was more readable than the previous guards. Guards make more sense when there are more than 2 possible branches. I can revert it and we can take the discussion on the style doc, however.

Reply:
It was better than the previous guards, indeed, but what about:
```
  case readMaybe addrStr of
￼	    Nothing -> Nothing
￼	    Just addr -> if val /= 0
￼	                   then Just $ NodeAddress addr port
￼	                   else Nothing
```
->
```
  case readMaybe addrStr of
    Nothing -> Nothing
    Just addr | val /= 0  -> Just $ NodeAddress addr port
              | otherwise -> Nothing
```

Reply:
Or maybe that just asks for a `fmap` with the conditional factored into a `where` binding?

Reply:
Going to go with:
```
case readMaybe addrStr of
    Nothing -> Nothing
￼   Just addr -> if val /= 0
                 then Just $ NodeAddress addr port
￼                else Nothing
```

Reply:
ok